### PR TITLE
Extend some doccomments based on Github discussions

### PR DESCRIPTION
**Pull request by vi**
Addresses:

* #1234
* #1235
* #1154

### CODE REVIEW

File: src/tree.rs [hunk 1]
```diff
@@ -335,6 +337,9 @@ impl Tree {
     /// assert_eq!(&processed.get(b"k3").unwrap().unwrap(), b"yappin' ligers");
     /// # Ok(()) }
     /// ```
+    ///
+    /// Note simple string prepeding in the example above should be natively changed
+    /// to some complex operation, as transactions may be tried multiple times.
```
Comment (spacejam) [inline src/tree.rs:342]: incorrect. transactions are serializable.

File: src/lib.rs [hunk 2]
```diff
@@ -8,6 +8,11 @@
 //! are supported with the
 //! [`Db::open_tree`](struct.Db.html#method.open_tree) method.
 //!
+//! Sled is not designed to be used by multiple processes simultaneously.
+//! Although Sled tries to use advisory filesystem locking to
+//! protect the database from accidental access from foreign
+//! processes, this should not be reiled upon.
```
Comment (spacejam) [inline src/lib.rs:14]: no need for this comment here

File: src/db.rs [hunk 3]
```diff
@@ -5,6 +5,9 @@ use crate::*;
 /// The `sled` embedded database! Implements
 /// `Deref<Target = sled::Tree>` to refer to
 /// a default keyspace / namespace / bucket.
+///
+/// Note that dropping `Db` does not gurantee closing the database. 
+/// Sled may still access the underlying directory under the hood.
```
Comment (spacejam) [inline src/db.rs:10]: no need for this comment here

File: src/tree.rs [hunk 4]
```diff
@@ -231,6 +231,8 @@ impl Tree {
 
     /// Perform a multi-key serializable transaction.
     ///
+    /// User-provided function `f` may be run multiple times in case of conflicts.
+    ///
```
Comment (spacejam) [inline src/tree.rs:235]: please get rid of all of the other comments except this one, which is the only one we discussed earlier.

File: src/lib.rs (hunk 2 above)
Comment (vi) [reply src/lib.rs:14]: Where should it be instead?

I don't see other mentions of the shared multiprocess use cases in the documentation, yet questions do arise.

File: src/db.rs (hunk 3 above)
Comment (vi) [reply src/db.rs:10]: Where should it be (assuming the comment itself is not wrong, just misplaced)?

Filesystem usage aspect may be important for users, especially, for example, if the same process manages mounts/unmounts of the filesystem that backs the database.

File: src/tree.rs (hunk 1 above)
Comment (vi) [reply src/tree.rs:342]: Shall I try to construct some concrete example (involving external I/O from within a transaction) where this break down?

**Comment by spacejam** [comment]
Apologies for my harsh tone. I was feeling a bit frustrated at the feeling that I have been repeating the same thing, but better docs are a great way to address this frustration. I've reworded a few of your comments and placed them in places I hope will be visible to people who will encounter them. https://github.com/spacejam/sled/commit/66cf95758faac3cfbf837796c4cf35aa926f3ea3

File: src/lib.rs (hunk 2 above)
Comment (spacejam) [reply src/lib.rs:14]: I've reworded this and placed it high up on the main README https://github.com/spacejam/sled/commit/66cf95758faac3cfbf837796c4cf35aa926f3ea3

**Pull request merged**